gbp/scripts/clone.py

```diff
@@ -208,6 +209,31 @@ def main(argv):
 
         repo_setup.set_user_name_and_email(options.repo_user, options.repo_email, repo)
 
+        if options.upstream_remote:
+            try:
+                with open(os.path.join(clone_to, 'debian', 'upstream', 'metadata')) as f:
+                    import yaml
+
+                    y = yaml.load(f, Loader=yaml.CSafeLoader)
+                    try:
+                        upstream_repo = y['Repository']
+
+                        try:
+                            # check if it exists as a git repo
+                            repo.fetch(repo=upstream_repo)
+
+                            repo.add_remote_repo(options.upstream_remote,
+                                                 upstream_repo,
+                                                 tags=True)
+                        except GitRepositoryError:
+                            gbp.log.warn(
+                                'Unable to fetch upstream repository %s, '
+                                'ignoring.' % upstream_repo)
+                    except KeyError:
+                        pass
+            except FileNotFoundError:
+                pass
```

Comment:
Could that go into a function? Also we don't have a way to fail cloning if no upstream remote exists or when cloning fails. I'd expect things to fail if that feature is enabled otherwise things become hard to use in scripts. Maybe we need a separate option to enable adding `upstream` with [yes|no|auto] (the later meaning add upstream if metadata exists)?

Reply:
Thing is we can't know from `debian/upstream/metadata` whether it is a git repository, a Mercurial one, SVN or anything else. So here I just try to clone it and don't really further interpret the failure since it could be "expected". Could log a debug message?

Reply:
Let's do heuristics then: So for --clone-upstream=auto :

- if it's salsa: git
- if it ends in .git: git
- if it's gitlab.com or github.com: git

if cloning fails we give an error. If it does not match the pattern ignore the error (and print a warning). We could go further and make this configurable (similar to what we have for `gbp create-remote-repo). For `--clone-upstream=yes` we always try to clone and fail if cloning fails.

And plesse add a code comment that points to the bug that says we need to be clearer on metadata (i think i saw this flying by)